control_panel_api/serializers.py

```diff
@@ -36,6 +36,14 @@ def update(self, instance, validated_data):
 
         return super().update(instance, validated_data)
 
+    def create(self, validated_data):
+        if validated_data['s3bucket'].is_data_warehouse:
+            raise serializers.ValidationError(
+                'Only app s3bucket can be related.'
```

Comment:
I would tweak the error message to that it's more explicit. Maybe something like `Apps cannot access data warehouse S3 buckets`, or something like that.